logger.go

```diff
@@ -31,46 +40,91 @@ func New(config Config) (*MicroLogger, error) {
 		config.IOWriter = DefaultIOWriter
 	}
 
-	kitLogger := kitlog.NewJSONLogger(kitlog.NewSyncWriter(config.IOWriter))
-
-	kitLogger = kitlog.With(
-		kitLogger,
-		"caller", config.Caller,
-		"time", config.TimestampFormatter,
-	)
+	log := log.New(config.IOWriter, "", 0)
 
-	l := &MicroLogger{
-		logger: kitLogger,
+	logger := &MicroLogger{
+		human: config.Human,
+		keyVals: map[interface{}]interface{}{
+			"caller": config.Caller,
+			"time":   config.TimestampFormatter,
+			"level":  "info",
+		},
+		log: log,
 	}
 
-	return l, nil
+	return logger, nil
+}
+
+func ToString(v interface{}) string {
+	valuer, ok := v.(Valuer)
+	if ok {
+		v = valuer()
+	}
+	return fmt.Sprintf("%v", v)
 }
 
 func (l *MicroLogger) Log(keyVals ...interface{}) error {
-	return l.logger.Log(keyVals...)
+	combined := map[string]string{}
+	for k, v := range l.keyVals {
+		k := ToString(k)
+		v := ToString(v)
+		combined[k] = v
+	}
+	for i := 0; i < len(keyVals); i += 2 {
+		k := ToString(keyVals[i])
+		v := ToString(keyVals[i+1])
+		combined[k] = v
+	}
+	var encoded string
+	if l.human {
+		level := strings.ToUpper(combined["level"])
+		switch level {
+		case "ERROR":
```

Comment:
Maybe we can reuse parts of the code over here?https://github.com/giantswarm/micrologger/blob/master/activation_logger.go#L14-L28

Reply:
I hadn't looked too much into `activation_logger` although I figured there would be some overlap. Thanks for pointing this out.